crates/hyperqueue/src/client/output/cli.rs

```diff
@@ -536,49 +541,52 @@ impl Output for CliOutput {
         );
     }
 
-    fn print_job_output(&self, job: JobDetail, task_selector: &Selector, output_stream: &OutputStream) -> anyhow::Result<()> {
+    fn print_job_output(
+        &self,
+        job: JobDetail,
+        task_selector: &Selector,
+        output_stream: &OutputStream,
+    ) -> anyhow::Result<()> {
         fn read_stream(task_state: &JobTaskState, output_stream: &OutputStream) {
             let (_, stdout, stderr) = get_task_paths(task_state);
             let stdin = match output_stream {
-                OutputStream::Stdout => {stdout}
-                OutputStream::Stderr => {stderr}
+                OutputStream::Stdout => stdout,
+                OutputStream::Stderr => stderr,
             };
 
-            let file = File::open(stdin).expect("Unable to open file");
-            let mut reader = BufReader::new(file);
-            for line in reader.lines() {
-                let line = line.expect("Unable to read line");
-                println!("{}", line);
-            }
+            match File::open(stdin) {
+                Ok(mut file) => {
+                    std::io::copy(&mut file, &mut std::io::stdout());
+                }
+                Err(_) => log::warn!("File: {} cannot be opened", stdin),
+            };
         }
 
-        let JobDetail {
-            info,
-            job_desc,
-            tasks,
-            max_fails: _,
-            submission_date,
-            completion_date_or_now,
-            submit_dir,
-        } = job;
-
         match task_selector {
             Selector::All => {
-                tasks.iter().for_each(|task| {
+                job.tasks.iter().for_each(|task| {
                     read_stream(&task.state, output_stream);
                 });
             }
             Selector::Specific(arr) => {
-                arr.iter().for_each(|id| {
-                    read_stream(&tasks[id.to_usize()].state, output_stream);
-                });
+                let tasks: Vec<(u32, Option<&JobTaskInfo>)> = arr
+                    .iter()
+                    .map(|id| (id, job.tasks.get(id.to_usize())))
+                    .collect();
+                for (id, task) in tasks {
+                    if task.is_none() {
+                        log::error!("Task {} not found", id);
+                        return Ok(());
```

Comment:
If a task is not found, let's just continue the loop (and print a warning, not an error).